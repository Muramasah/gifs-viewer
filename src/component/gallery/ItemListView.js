//Libraries
import React from 'react';
//Components
import EmptyGallery from './EmptyGallery';
import NoneColumnGapDiv from './presenter/NoneColumnGapDiv';
//Helpers
import { isObjectEmptyOrUndefined } from '../../helper/object';
import { generateUiID } from '../../helper/dom';

const ItemListView = (props) => {
    const { emptyMessage, gifs, ImageItem, onClickButton } = props;

    if (isObjectEmptyOrUndefined(gifs)) {
        return <EmptyGallery emptyMessage={emptyMessage} />
    } else {
        return (
            <NoneColumnGapDiv>
                {Object.keys(gifs).map((id, index) => {
                    const gif = gifs[id];
                    const { slug, url } = gif;
                    const uiID = generateUiID();

                    return <ImageItem
                        gif={gif}
                        key={index}
                        id={id}
                        uiID={uiID}
                        imageUrl={url}
                        slug={slug}
                        onClickButton={onClickButton}
                    />
                })}
            </NoneColumnGapDiv>
        );
    }
}

export default ItemListView;