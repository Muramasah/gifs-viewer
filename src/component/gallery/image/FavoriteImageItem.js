//Libraries
import React from 'react';
//Components
import StyledImageItem from './StyledImageItem';
import DislikeButton from './button/DislikeButton';

const FavoriteImageItem = ({ imageUrl, slug, onLoad, onDislike }) => {
    return (
        <div>
            <StyledImageItem
                src={imageUrl}
                alt={slug}
                onLoad={onLoad}
            />
            <DislikeButton onDislike={onDislike} />
        </div>
    );
}

export default FavoriteImageItem;