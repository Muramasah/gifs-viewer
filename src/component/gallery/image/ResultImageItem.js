//Libraries
import React from 'react';
//Components
import StyledImageItem from './StyledImageItem';
import ResultImageButton from './button/ResultImageButton';

const ResultImageItem = ({ imageUrl, slug, isFavorite, onLoad, onClickImage, onLike }) => {
    return (
        <div>
            <StyledImageItem
                src={imageUrl}
                alt={slug}
                onLoad={onLoad}
                onClick={onClickImage}
            />
            <ResultImageButton
                isFavorite={isFavorite}
                onLike={onLike}
            />
        </div>
    );
}

export default ResultImageItem;