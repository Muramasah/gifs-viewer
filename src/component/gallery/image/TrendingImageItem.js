//Libraries
import React from 'react';
//Components
import StyledImageItem from './StyledImageItem';
import IsFavoriteButton from './button/IsFavoriteButton';

const TrendingImageItem = ({ imageUrl, slug, onLoad }) => {
    return (
        <div>
            <StyledImageItem onLoad={event => onLoad} src={imageUrl} alt={slug} />
            <IsFavoriteButton />
        </div>
    );
}

export default TrendingImageItem;