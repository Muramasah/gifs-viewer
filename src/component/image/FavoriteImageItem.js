//Libraries
import React from 'react';
//Components
import ImageGalleryItem from './ImageGalleryItem';
import DislikeButton from '../button/DislikeButton';
import InlineDiv from './presenter/InlineDiv';

const FavoriteImageItem = (props) => {
    const { gif, uiID, imageUrl, slug, onClickButton } = props;

    return (
        <InlineDiv>
            <ImageGalleryItem
                uiID={uiID}
                src={imageUrl}
                alt={slug}
                gif={gif}
                onClickButton={onClickButton}
                ItemButton={DislikeButton}
            />
        </InlineDiv>
    );
}

export default FavoriteImageItem;