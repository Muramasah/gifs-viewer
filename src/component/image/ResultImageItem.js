//Libraries
import React from 'react';
//Components
import ImageGalleryItem from './ImageGalleryItem';
import ResultImageButton from '../button/ResultImageButton';
import InlineDiv from './presenter/InlineDiv';

const ResultImageItem = (props) => {
    const { gif, uiID, imageUrl, slug, onClickButton } = props;

    return (
        <InlineDiv>
            <ImageGalleryItem
                uiID={uiID}
                src={imageUrl}
                alt={slug}
                gif={gif}
                onClickButton={onClickButton}
                ItemButton={ResultImageButton}
            />
        </InlineDiv>
    );
}

export default ResultImageItem;