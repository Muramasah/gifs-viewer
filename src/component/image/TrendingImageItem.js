//Libraries
import React from 'react';
//Components
import ImageGalleryItem from './ImageGalleryItem';
import InlineDiv from './presenter/InlineDiv';

const TrendingImageItem = (props) => {
    const { uiID, imageUrl, slug } = props;

    return (
        <InlineDiv>
            <ImageGalleryItem
                uiID={uiID}
                src={imageUrl}
                alt={slug}
            />
        </InlineDiv >
    );
}

export default TrendingImageItem;