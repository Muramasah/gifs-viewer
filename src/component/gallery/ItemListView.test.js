import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ItemListView from './ItemListView';
import ResultImageItem from '../image/ResultImageItem';
import TrendingImageItem from '../image/TrendingImageItem';
import FavoriteImageItem from '../image/FavoriteImageItem';

configure({ adapter: new Adapter() });

jest.mock('../../helper/dom'); // this happens automatically with automocking
const generateUiID = require('../../helper/dom').generateUiID;
generateUiID.mockImplementation(() => 'poU45wTKf1Qut1XirnOq');

const gifs = {
    '3ohc123V1Dr1nf8lcQ': {
        id: '3ohc123V1Dr1nf8lcQ',
        slug: 'dance-cute-bear-3ohc123V1Dr1nf8lcQ',
        url: 'https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif'
    }
}

test('ItemListView renders with ResultImageButton without crashing', () => {
    const itemListView = shallow(<ItemListView
        emptyMessage='empty'
        gifs={gifs}
        ImageItem={ResultImageItem}
        onClickButton={() => { }}
    />);

    expect(itemListView).toMatchSnapshot();
});

test('ItemListView renders with TrendingImageItem and empty gif object without crashing', () => {
    const itemListView = shallow(<ItemListView
        emptyMessage='empty'
        gifs={{}}
        ImageItem={TrendingImageItem}
        onClickButton={() => { }}
    />);

    expect(itemListView).toMatchSnapshot();
});

test('ItemListView renders with FavoriteImageItem without crashing', () => {
    const itemListView = shallow(<ItemListView
        emptyMessage='empty'
        gifs={gifs}
        ImageItem={FavoriteImageItem}
        onClickButton={() => { }}
    />);

    expect(itemListView).toMatchSnapshot();
});