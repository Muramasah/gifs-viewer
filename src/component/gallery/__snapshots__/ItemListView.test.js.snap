// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemListView renders with FavoriteImageItem without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemListView
    ImageItem={[Function]}
    emptyMessage="empty"
    gifs={
        Object {
            "3ohc123V1Dr1nf8lcQ": Object {
              "id": "3ohc123V1Dr1nf8lcQ",
              "slug": "dance-cute-bear-3ohc123V1Dr1nf8lcQ",
              "url": "https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif",
            },
          }
    }
    onClickButton={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <FavoriteImageItem
          gif={
                    Object {
                              "id": "3ohc123V1Dr1nf8lcQ",
                              "slug": "dance-cute-bear-3ohc123V1Dr1nf8lcQ",
                              "url": "https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif",
                            }
          }
          id="3ohc123V1Dr1nf8lcQ"
          imageUrl="https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif"
          onClickButton={[Function]}
          slug="dance-cute-bear-3ohc123V1Dr1nf8lcQ"
          uiID="poU45wTKf1Qut1XirnOq"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "gif": Object {
            "id": "3ohc123V1Dr1nf8lcQ",
            "slug": "dance-cute-bear-3ohc123V1Dr1nf8lcQ",
            "url": "https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif",
          },
          "id": "3ohc123V1Dr1nf8lcQ",
          "imageUrl": "https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif",
          "onClickButton": [Function],
          "slug": "dance-cute-bear-3ohc123V1Dr1nf8lcQ",
          "uiID": "poU45wTKf1Qut1XirnOq",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <FavoriteImageItem
            gif={
                        Object {
                                    "id": "3ohc123V1Dr1nf8lcQ",
                                    "slug": "dance-cute-bear-3ohc123V1Dr1nf8lcQ",
                                    "url": "https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif",
                                  }
            }
            id="3ohc123V1Dr1nf8lcQ"
            imageUrl="https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif"
            onClickButton={[Function]}
            slug="dance-cute-bear-3ohc123V1Dr1nf8lcQ"
            uiID="poU45wTKf1Qut1XirnOq"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "gif": Object {
              "id": "3ohc123V1Dr1nf8lcQ",
              "slug": "dance-cute-bear-3ohc123V1Dr1nf8lcQ",
              "url": "https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif",
            },
            "id": "3ohc123V1Dr1nf8lcQ",
            "imageUrl": "https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif",
            "onClickButton": [Function],
            "slug": "dance-cute-bear-3ohc123V1Dr1nf8lcQ",
            "uiID": "poU45wTKf1Qut1XirnOq",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ItemListView renders with ResultImageButton without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemListView
    ImageItem={[Function]}
    emptyMessage="empty"
    gifs={
        Object {
            "3ohc123V1Dr1nf8lcQ": Object {
              "id": "3ohc123V1Dr1nf8lcQ",
              "slug": "dance-cute-bear-3ohc123V1Dr1nf8lcQ",
              "url": "https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif",
            },
          }
    }
    onClickButton={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ResultImageItem
          gif={
                    Object {
                              "id": "3ohc123V1Dr1nf8lcQ",
                              "slug": "dance-cute-bear-3ohc123V1Dr1nf8lcQ",
                              "url": "https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif",
                            }
          }
          id="3ohc123V1Dr1nf8lcQ"
          imageUrl="https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif"
          onClickButton={[Function]}
          slug="dance-cute-bear-3ohc123V1Dr1nf8lcQ"
          uiID="poU45wTKf1Qut1XirnOq"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "gif": Object {
            "id": "3ohc123V1Dr1nf8lcQ",
            "slug": "dance-cute-bear-3ohc123V1Dr1nf8lcQ",
            "url": "https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif",
          },
          "id": "3ohc123V1Dr1nf8lcQ",
          "imageUrl": "https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif",
          "onClickButton": [Function],
          "slug": "dance-cute-bear-3ohc123V1Dr1nf8lcQ",
          "uiID": "poU45wTKf1Qut1XirnOq",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ResultImageItem
            gif={
                        Object {
                                    "id": "3ohc123V1Dr1nf8lcQ",
                                    "slug": "dance-cute-bear-3ohc123V1Dr1nf8lcQ",
                                    "url": "https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif",
                                  }
            }
            id="3ohc123V1Dr1nf8lcQ"
            imageUrl="https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif"
            onClickButton={[Function]}
            slug="dance-cute-bear-3ohc123V1Dr1nf8lcQ"
            uiID="poU45wTKf1Qut1XirnOq"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "gif": Object {
              "id": "3ohc123V1Dr1nf8lcQ",
              "slug": "dance-cute-bear-3ohc123V1Dr1nf8lcQ",
              "url": "https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif",
            },
            "id": "3ohc123V1Dr1nf8lcQ",
            "imageUrl": "https://media1.giphy.com/media/3ohc123V1Dr1nf8lcQ/giphy.gif",
            "onClickButton": [Function],
            "slug": "dance-cute-bear-3ohc123V1Dr1nf8lcQ",
            "uiID": "poU45wTKf1Qut1XirnOq",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ItemListView renders with TrendingImageItem and empty gif object without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemListView
    ImageItem={[Function]}
    emptyMessage="empty"
    gifs={Object {}}
    onClickButton={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "emptyMessage": "empty",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "emptyMessage": "empty",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
